server/tests/post-service.test.ts

```diff
@@ -0,0 +1,136 @@
+import { mock, instance, when, verify, deepEqual, spy, resetCalls } from 'ts-mockito';
+import { Dependencies } from '../types';
+import { Repository, DeleteResult, UpdateResult } from 'typeorm';
+import { Post } from '../entities/post';
+import { PostService } from '../service/post-service';
+import { User } from '../entities/user';
+import { UserService } from '../service/user-service';
+
+const postRepository = <Repository<Post>>mock(Repository);
+const repoInstance = instance(postRepository);
+const service = new PostService(<Dependencies>{ postRepository: repoInstance });
```

Comment:
Pick if you use bracket syntax or `as` syntax when doing castings :D Not good idea to mix both :D